### DISCUSSION TITLE
In a Github Actions automated release, where is the best place to store GPG secrets etc?

### DISCUSSION THREAD

**Discussion by ge0ffrey** [jreleaser/jreleaser#1396] [General]
To be able to release to Maven central, JReleaser needs a number of secrets, such as:

- the GPG private key
- the sonatype nexus 2 password

When triggered from a GitHub Actions, it's possible to provide these secrets through "Actions secrets", but that means **anyone running the workflow knows and uses the same e-mail and sonatype password**:

![image](https://github.com/jreleaser/jreleaser/assets/176880/d18ce22d-bba8-4b07-8b51-595569db5b83)

In practice, this means a "release account" needs to be created.

Is there a way to use the GPG private key and sonatype password from the person that triggers the GitHub action instead? A personal github profile supports storing GPG secrets.

**Comment by aalmiray** [comment]
It seems go me this is more of a GitHub question than JReleaser itself. That being said, there are 3rd party vault/secrets services that can be integrated with GitHub, providing the behavior you seek.

**Reply by ge0ffrey** [reply]
Good point. I agree that it's more of a GitHub question, although I would imagine many jrelease users face this dilemma. Maybe I should ask my question differently: what do you and the people here think is the recommended way? And why?

**Reply by aalmiray** [reply]
Unless GitHub offers an option to fetch secrets per account (which I'm unaware at the moment) you'd have to configure a 3rd party vault service that can provide the behavior you seek.
